### PR TITLE
Minor slicing fix for certain areas

### PR DESCRIPTION
**Pull request by adahn6**
### Description:

In rare cases, an Area can get sliced into resulting multipolygon geometry; this corner case arises when an Area completely contains a boundary, leaving one or more holes after slicing. Since Areas only support simply polygon geometry, the code path rejects this new sliced geometry and attempt to convert the Area to a linear Line geometry and slice that. However, in this case, using `Area.asPolygon()` in the new CompleteLine accidentally preserves the polygonal geometry. So instead, a fix here casts it to a JTS Polygon, then converts the exterior ring to a PolyLine, which guarantees a linear representation when sliced again.
 
### Potential Impact:

Extremely limited, this is an unusual corner case.

### Test Results:

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

**Review by lucaspcram — APPROVED** [review]

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java [hunk 1]
```diff
@@ -1319,7 +1319,9 @@ private void sliceArea(final Area area)
             final Set<Long> relationIds = new HashSet<>();
             area.relations().forEach(relation -> relationIds.add(relation.getIdentifier()));
             final CompleteLine lineFromArea = new CompleteLine(area.getIdentifier(),
-                    area.asPolygon(), area.getTags(), relationIds);
+                    JTS_POLYLINE_CONVERTER.backwardConvert(
```
Comment (MikeGost) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java:1322]: Is it worth adding a test case for this edge case?

Comment (adahn6) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java:1322]: probably 💀for now I'm prioritizing a fix since this is a one-in-a-million corner case, but I'll see if I can capture the behavior in a sufficient unit test
